src/main/java/com/github/dockerjava/core/command/CreateContainerCmdImpl.java

```diff
@@ -529,6 +530,13 @@ public CreateContainerCmd withAttachStdout(Boolean attachStdout) {
         return this;
     }
 
+    @Override
+    public CreateContainerCmd withAutoRemove(Boolean autoRemove) {
```

Comment:
remove this chain, you are adding method for hostconfig, not for createcontainerCmdImpl

Reply:
I am sorry but I don't understand your request. I mean that right now, all other parameters of HostConfig can be set following a chain of setters in both CreateContainerCmd/Impl and in HostConfig.  You can take a look at _withCapDrop_, _withBinds_, _withDevices_ and so on.  Do you want that _autoRemove_  attribute would be different to all others? Why should this attribute be different? What do I misunderstand?

Reply:
because it's old code that is keeping for backward compatibility. It's nightmware to proxy(chain) all possible fields via one class, docker-java is tied to api, not to CLI. So just remove redundant chain for new code.

Reply:
OK. I understood it!
What do you think about adding a @Deprecated annotation to all these "old" methods, to avoid coding in this unwanted way? Maybe @silvestre (#1052) and I would not have done the same! : smiley:

Reply:
I thought i documented it somewhere. In future methods will be just removed.